src/modules/commander/ManualControl.cpp

```diff
@@ -0,0 +1,194 @@
+			(fabsf(_manual_control_setpoint.z - _last_manual_control_setpoint.z) * 2.f > minimum_stick_change);
+		const bool use_throttle = !(_param_rc_override.get() & OverrideBits::OVERRIDE_IGNORE_THROTTLE_BIT);
+
+		if (rpy_moved || (use_throttle && throttle_moved)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+bool ManualControl::wantsDisarm(const vehicle_control_mode_s &vehicle_control_mode,
+				const vehicle_status_s &vehicle_status,
+				manual_control_switches_s &manual_control_switches, const bool landed)
+{
+	bool ret = false;
+
+	// no switch or button is mapped
+	const bool use_stick = manual_control_switches.arm_switch == manual_control_switches_s::SWITCH_POS_NONE;
+	// arm switch mapped and "switch is button" configured
+	const bool use_button = !use_stick && _param_com_arm_swisbtn.get();
+	// arm switch mapped and "switch_is_button" configured
+	const bool use_switch = !use_stick && !use_button;
+
+	const bool armed = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED);
+	const bool stick_in_lower_left = use_stick
+					 && isThrottleLow()
+					 && _manual_control_setpoint.r < -.9f;
+	const bool arm_button_pressed = (manual_control_switches.arm_switch == manual_control_switches_s::SWITCH_POS_ON)
+					&& use_button;
+	const bool arm_switch_to_disarm_transition = use_switch
+			&& (_last_manual_control_switches_arm_switch == manual_control_switches_s::SWITCH_POS_ON)
+			&& (manual_control_switches.arm_switch == manual_control_switches_s::SWITCH_POS_OFF);
+	const bool mc_manual_thrust_mode = vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+					   && vehicle_control_mode.flag_control_manual_enabled
+					   && !vehicle_control_mode.flag_control_climb_rate_enabled;
+
+	if (armed
+	    && (landed || mc_manual_thrust_mode)
+	    && (stick_in_lower_left || arm_button_pressed || arm_switch_to_disarm_transition)) {
+
+		const bool last_disarm_hysteresis = _stick_disarm_hysteresis.get_state();
+		_stick_disarm_hysteresis.set_state_and_update(true, hrt_absolute_time());
+		const bool disarm_trigger = !last_disarm_hysteresis && _stick_disarm_hysteresis.get_state()
+					    && !_stick_arm_hysteresis.get_state();
+
+		if (disarm_trigger || arm_switch_to_disarm_transition) {
+			ret = true;
+		}
+
+	} else if (!arm_button_pressed) {
+
+		_stick_disarm_hysteresis.set_state_and_update(false, hrt_absolute_time());
+	}
+
+	return ret;
+}
+
+bool ManualControl::wantsArm(const vehicle_control_mode_s &vehicle_control_mode, const vehicle_status_s &vehicle_status,
+			     manual_control_switches_s &manual_control_switches, const bool landed)
+{
+	bool ret = false;
+
+	// no switch or button is mapped
+	const bool use_stick = manual_control_switches.arm_switch == manual_control_switches_s::SWITCH_POS_NONE;
+	// arm switch mapped and "switch is button" configured
+	const bool use_button = !use_stick && _param_com_arm_swisbtn.get();
+	// arm switch mapped and "switch_is_button" configured
```

Comment:
```suggestion
	// arm switch mapped and "switch_is_button" not configured
```
not sure if the comment is useful

Reply:
Probably they just helped me develop. The variable names are clear and the conditions aren't crazy. Removed them.

Reply:
d909125